tools/ci_build/github/azure-pipelines/templates/clean-agent-build-directory-step.yml

```diff
@@ -1,9 +1,5 @@
-# cleans $(Agent.BuildDirectory) unconditionally on multiple platforms
-
 steps:
-- script: rd /S /Q $(Agent.BuildDirectory)
-  displayName: Clean build files (Windows)
-  condition: eq(variables['Agent.OS'], 'Windows_NT') # and always()
-- script: sudo rm -rf $(Agent.BuildDirectory)
-  displayName: Clean build files (POSIX)
-  condition: not(eq(variables['Agent.OS'], 'Windows_NT')) # and always()
+  - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
```

Comment:
For this to clean everything up, we need to set the build definition options appropriately (clean = true, clean options = "All build directories"). I don't know if that is supported from YAML (see https://github.com/Microsoft/azure-pipelines-agent/issues/1530#issuecomment-438855783 - setting workspace.clean did not seem to work).

I think it would be preferable to have this clean step always clean everything up instead of relying on other build settings, especially if we have to configure them outside of the YAML.

Reply:
We should set correct clean options, otherwise we may get wrong build result.
The cleanup step at the end of build is just for saving disk space.